src/NuGet.Clients/NuGet.PackageManagement.UI/Options/ConfigurationFilesViewModel.cs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Windows.Input;
+using Microsoft.VisualStudio.ComponentModelHost;
+using Microsoft.VisualStudio.PlatformUI;
+using Microsoft.VisualStudio.Services.Common;
+using NuGet.ProjectManagement;
+using NuGet.VisualStudio;
+
+namespace NuGet.PackageManagement.UI.Options
+{
+    public class ConfigurationFilesViewModel
+    {
+        public ObservableCollection<string> ConfigurationFilesCollection { get; private set; }
+        public string SelectedPath { get; set; }
+        public ICommand OpenConfigurationFile { get; set; }
+
+        public ConfigurationFilesViewModel()
+        {
+            ConfigurationFilesCollection = new ObservableCollection<string>();
+            OpenConfigurationFile = new DelegateCommand(ExecuteOpenConfigurationFile, IsSelectedPath, NuGetUIThreadHelper.JoinableTaskFactory);
+        }
+
+        private bool IsSelectedPath()
+        {
+            return SelectedPath != null;
+        }
+
+        private void ExecuteOpenConfigurationFile()
+        {
+            OpenConfigFile(SelectedPath);
+        }
+        internal void InitializeOnActivated(CancellationToken cancellationToken)
+        {
+            SetConfigPaths();
+        }
+
+        public void SetConfigPaths()
+        {
+            IComponentModel componentModelMapping = NuGetUIThreadHelper.JoinableTaskFactory.Run(ServiceLocator.GetComponentModelAsync);
```

Comment:
1. using `NuGetUIThreadHelper.JoinableTaskFactory` basically makes this untestable.
2. The component model is used twice, here in `SetConfigPaths`, and again in `OpenConfigFile`, but the component model is itself a singleton. Therefore, it could be obtained once, passed to this class's constructor. This has the side effect that `IComponentModel` can be mocked and therefore tests can be written.